blaze/expr/tests/test_collections.py

```diff
@@ -108,13 +108,21 @@ def test_join_option_types():
     assert join(b, a, 'x').dshape == dshape('var * {x: int}')
 
 
-def test_join_mismatched_schema():
+def test_join_exceptions():
+    'mismatched schema; no shared fields'
     a = symbol('a', 'var * {x: int}')
-    b = symbol('b', 'var * {x: string}')
-
     with pytest.raises(TypeError):
+        b = symbol('b', 'var * {x: string}')
```

Comment:
We should move this out of the context manager. As a general rule context invariant code should not be placed in the context manager when possible because then readers need to know if the context manager affects the code in any way. In this particular case, if `symbol` raises a `TypeError` then this test will pass when it shouldn't